src/containers/TemplateBuilder/template/Actions/ActionConfig.tsx

```diff
@@ -40,17 +40,21 @@ const getState: GetState = (action: TemplateAction) => ({
 })
 
 const ActionConfig: React.FC<ActionConfigProps> = ({ templateAction, onClose }) => {
+  const { strings } = useLanguageProvider()
   const {
     template: { id: templateId, isDraft },
   } = useTemplateState()
   const { updateTemplate } = useOperationState()
   const [state, setState] = useState<ActionUpdateState | null>(null)
   const { allActionsByCode, applicationData } = useActionState()
+  const [shouldUpdate, setShouldUpdate] = useState<boolean>(false)
+  const [changesSaved, setChangesSaved] = useState<boolean>(false)
```

Comment:
No, you might have saved and didn't close the window. Then changed something else:
- `shouldUpdate` = true, `changesSaved` = false (because you don't see the message yet)

After you click on the Save button, now they are:
- `shouldUpdate` = false, `changesSaved` = true

And before you make any changes, both are false, they are never both `true` at the same time...